### PR TITLE
 OS-8584 Introduce gcc14 as the shadow compiler (fix gcc14 primary) @danmcd

### PR DESCRIPTION
**Pull request by danmcd**
Multiple components, through trial and error.  It now builds.  I cannot guarantee its usefulness.

Before we enable gcc14 as primary, we must inspect all of the components' `configure` outputs compared with the more-lenient gcc10 build.  We wanna make sure we're not failing feature checks because of stupidity in the `configure` scripts.  You'll notice many such changes in this batch as-is.

### CODE REVIEW

File: socat/Patches/gcc10-gcc14.patch [hunk 1]
```diff
@@ -34,3 +34,14 @@ diff -ru a/fdname.c b/fdname.c
  #elif HAVE_GETPROTOBYNUMBER
     protoentp = getprotobynumber(proto);
     strncpy(protoname, protoentp->p_name, sizeof(protoname));
+diff -ru a/sysincludes.h b/sysincludes.h
+--- a/sysincludes.h	Sun Jan  3 13:23:22 2021
++++ b/sysincludes.h	Thu Nov 21 17:57:37 2024
+@@ -162,6 +162,7 @@
+ #endif
+ #if HAVE_SYS_STROPTS_H
+ #include <sys/stropts.h>	/* SunOS I_PUSH ... */
```
Comment (tsoome) [inline socat/Patches/gcc10-gcc14.patch:43]: small nit: with <stropts.h> we do not really need <sys/stropts.h>

**Review by tsoome — APPROVED** [review]
2 notes, but imo good to go.

**Comment by danmcd** [comment]
Testing notes for this are in the ticket.  I don't think these will harm the `gcc10` build. If you think I'm missing something please hold me to it.

File: socat/Patches/gcc10-gcc14.patch (hunk 1 above)
Comment (danmcd) [reply socat/Patches/gcc10-gcc14.patch:43]: I wasn't sure.  Won't fix for now.

File: bash/Patches/fix-configure-gcc14.diff [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+ 
+Common subdirectories: a/lib and b/lib
+diff -ru a/builtins/command.def b/builtins/command.def
+--- a/builtins/command.def	Thu Nov 21 13:16:59 2024
++++ b/builtins/command.def	Thu Nov 21 13:23:50 2024
+@@ -59,6 +59,8 @@
+ #if defined (_CS_PATH) && defined (HAVE_CONFSTR) && !HAVE_DECL_CONFSTR
+ extern size_t confstr __P((int, char *, size_t));
+ #endif
++/* XXX SmartOS - confstr isn't defined in spite of above!!! Fix! */
```
[the remark this answers is not in the document]
Comment (danmcd) [reply bash/Patches/fix-configure-gcc14.diff:266]: Ooof.  Still gonna keep this here, however, and we can nuke it when we bump the bash version.

**Pull request merged**